compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrInlineCodegen.kt

```diff
@@ -42,20 +43,20 @@ class IrInlineCodegen(
     }
 
     override fun genValueAndPut(
-        irValueParameter: IrValueParameter?,
+        irValueParameter: IrValueParameter,
         argumentExpression: IrExpression,
         parameterType: Type,
-        parameterIndex: Int,
         codegen: ExpressionCodegen,
         blockInfo: BlockInfo
     ) {
-        if (irValueParameter?.isInlineParameter() == true && isInlineIrExpression(argumentExpression)) {
-            val irReference: IrFunctionReference =
-                (argumentExpression as IrBlock).statements.filterIsInstance<IrFunctionReference>().single()
-            rememberClosure(irReference, parameterType, irValueParameter) as IrExpressionLambdaImpl
-        } else {
-            putValueOnStack(argumentExpression, parameterType, irValueParameter?.index ?: -1, blockInfo)
+        if (irValueParameter.isInlineParameter() && isInlineIrExpression(argumentExpression)) {
```

Comment:
Please pass flag here: is parameter extensionReceiver or not, and check it in 'if'.
Now non-null 'irValueParameter' is passed and inliner tries to inline extension receiver argument as lambda one

Reply:
Is this the only situation where `isInlineParameter` and `isInlineIrExpression` are both satisfied and yet the argument does not have the right shape?

Assuming that this is the case, I simplified the code by moving the check to the `isInlineParameter` function.

Reply:
Yes, receiver inlining is not supported in old backend and in IR for now

Reply:
Good to know! Fixed.